serde/src/java/org/apache/hadoop/hive/serde2/WriteBuffers.java

```diff
@@ -149,7 +149,12 @@ public int unsafeHashCode(long offset, int length) {
   }
 
   public int hashCode(long offset, int length, Position readPos) {
+    // If caller has not set the read position, then set it.
     setReadPoint(offset, readPos);
+    return hashCode(length, readPos);
```

Comment:
Shall we move the positioning code to the unsafeHashCode method and remove this method? Looks like the only place its needed